libcodechecker/server/client_db_access_handler.py

```diff
@@ -93,13 +105,111 @@ def bugreportpoint_db_to_api(brp):
         fileId=brp.file_id)
 
 
+def unzip(workspace, b64zip):
+    '''This function unzips the base64 encoded zip file which should contain
+    exactly one directory. This directory is extracted to the given workspace.
+    The function returns the name of the extracted directory.'''
+
+    _, zip_file = tempfile.mkstemp('.zip', dir=workspace)
+
+    with open(zip_file, 'wb') as zip_f:
+        zip_f.write(base64.b64decode(b64zip))
+
+    zipf = zipfile.ZipFile(zip_file, 'r')
+
+    # A not too nice way to find the only directory in the zip archive.
+    first_file = zipf.namelist()[0]
+    zip_dir = first_file[:first_file.find('/')]
+
+    zipf.extractall(workspace)
+    zipf.close()
+
+    os.remove(zip_file)
+
+    return os.path.join(workspace, zip_dir)
+
+
+class StorageSession:
```

Comment:
Does introducing this StorageSession worth the benefits? Wouldn't it work if we keep our transaction scheme as it is for now and only change it in a subsequent pull request if necessary? What are the benefits of this scheme?